### PR TITLE
feature: first stab at merge-on-green

### PR DESCRIPTION
**Pull request by sofisl**
First stab at merge-on-green, using separate branch so as to not confuse w/ dev env

This PR:

(1) responds to PRs, identifies the owner, repo and branch for a specific PR
(2) identifies required status checks and/or config files with required status checks
(3) posts a status check that passes if there are any required status checks or config files, or fails if there aren't (also fails if there are less than 3 required status checks, or if status checks do not match up with config file)

### CODE REVIEW

File: packages/merge-on-green/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# merge-on-green
```
Comment (bcoe) [inline packages/merge-on-green/README.md]: this is looking good, we should make sure to upstream these template changes to your generator.

File: packages/merge-on-green/src/merge-on-green.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+          }
+        });
+      }
+    }
+
+    await context.github.checks.create(checkParams);
+  });
+};
+
+//TODO: fail if config and no branch protection, ask user to enforce branch protection
```
Comment (bcoe) [inline packages/merge-on-green/src/merge-on-green.ts]: I think we can remove these TODO comments now?

File: packages/merge-on-green/test/merge-on-green.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    );
+
+    it('gets branch information when a pull request is opened, responds with a status check', async () => {
+      const payload = require(resolve(
+        fixturesPath,
+        'events',
+        'pull_request_opened'
+      ));
+
+      // const scope = nockGetBranchProtection();
```
Comment (bcoe) [inline packages/merge-on-green/test/merge-on-green.ts]: we should probably get rid of this comment.

File: packages/merge-on-green/test/merge-on-green.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      const scope = nock('https://api.github.com')
+        .log(console.log)
+        .get('/repos/testOwner/testRepo/contents/.github/merge-on-green.yml')
+        .reply(200, { content: config.toString('base64') })
+        .get('/repos/testOwner/testRepo/branches/master/protection')
+        .reply(200)
+        .get('/repos/testOwner/testRepo/pulls/6/commits?per_page=100')
+        .reply(200, commits)
+        .post('/repos/testOwner/testRepo/check-runs', body => {
+          snapshot(body);
```
Comment (bcoe) [inline packages/merge-on-green/test/merge-on-green.ts]: we should check in the snapshot we get from this call.

File: packages/merge-on-green/test/merge-on-green.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      ));
+
+      // const scope = nockGetBranchProtection();
+
+      const scope = nock('https://api.github.com')
+        .log(console.log)
+        .get('/repos/testOwner/testRepo/contents/.github/merge-on-green.yml')
+        .reply(200, { content: config.toString('base64') })
+        .get('/repos/testOwner/testRepo/branches/master/protection')
+        .reply(200)
```
Comment (bcoe) [inline packages/merge-on-green/test/merge-on-green.ts]: It's probably worth adding a test for the bad case too.

File: packages/merge-on-green/README.md (hunk 1 above)
Comment (sofisl) [reply packages/merge-on-green/README.md]: ah right, will do

**Pull request merged**